### PR TITLE
Lg/dcos 50125 public ip cli

### PR DESCRIPTION
**Pull request by lsgunnlsgunn**
## Description

https://jira.mesosphere.com/browse/DCOS-50125
https://jira.mesosphere.com/browse/DCOS-51291
https://jira.mesosphere.com/browse/DCOS-50105
https://jira.mesosphere.com/browse/DCOS-44697
https://jira.mesosphere.com/browse/DCOS-49987
https://jira.mesosphere.com/browse/DCOS-28127

## Urgency
- [ ] Blocker 
- [X] High
- [ ] Medium

## Requirements
- Test all commands and procedures.
- Add [redirects](https://github.com/mesosphere/dcos-docs-site/wiki/Redirects).
- Change all affected versions (e.g. 1.7, 1.8, 1.9, 1.10, 1.11, 1.12).
- See the [contribution guidelines](https://github.com/mesosphere/dcos-docs-site/wiki/Contributing).

### CODE REVIEW

**Comment by lsgunnlsgunn** [comment]
Here is the initial DRAFT. I think the API section (in particular) needs technical vetting as it seems a bit more confusing (or possibly flat-out incorrect) than the CLI and GUI implementation for the feature. 
This feature has ripple effect in that we might want to replace the xref with the actual command in load balancing doc for example.

File: pages/1.13/cli/command-reference/dcos-node/dcos-node-list/index.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
```
Comment (ArmandGrillet) [inline pages/1.13/cli/command-reference/dcos-node/dcos-node-list/index.md]: This page doesn't need to be created in this PR, we're already creating it in https://github.com/mesosphere/dcos-docs-site/pull/2107 as mentioned in https://jira.mesosphere.com/browse/DCOS-50125?focusedCommentId=253016&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-253016

File: pages/1.13/administering-clusters/locate-public-agent/index.md [hunk 2]
```diff
@@ -6,23 +6,193 @@ menuWeight: 3
...
+    ```bash
+    http://luxi-sf7-elasticl-u70m3un6kcab-1100943753.us-west-2.elb.amazonaws.com/net/v1/nodes
+    ```
+
+1. Review the command output.
+
+    This API call returns information similar to the following in an unfiltered format:
+
+    ```bash
+    [{"updated":"2019-01-07T22:22:22.171Z","public_ips":["34.212.37.79"],"private_ip":"10.0.6.210","hostname":"ip-10-0-6-210"},{"updated":"2019-01-07T22:22:22.119Z","public_ips":["52.25.254.97"],"private_ip":"10.0.6.181","hostname":"ip-10-0-6-181"},{"updated":"2019-01-07T22:21:09.585Z","public_ips":["54.218.23.75"],"private_ip":"10.0.6.148","hostname":"ip-10-0-6-148"},{"updated":"2019-01-07T22:22:28.582Z","public_ips":[],"private_ip":"10.0.1.139","hostname":"ip-10-0-1-139"},{"updated":"2019-01-07T22:22:28.649Z","public_ips":[],"private_ip":"10.0.0.138","hostname":"ip-10-0-0-138"}]
```
Comment (jkoelker) [inline pages/1.13/administering-clusters/locate-public-agent/index.md]: nit: I think formatting all the json (though `jq` or similar) would help with readability. I always tend to skip `json` blobs that arn't formatted, if I find out later that it does have info in it I need to understand, copy it to a terminal to format it. Perhaps above have a note that raw api responses are in compact form and have been formatted in the documentation for readability or something like that.

File: pages/1.13/administering-clusters/locate-public-agent/index.md [hunk 3]
```diff
@@ -6,23 +6,193 @@ menuWeight: 3
...
+    ```
+
+    In both of these examples, there are three public IP addresses, one for each of the public agent nodes:
+    - 34.212.37.79
+    - 52.25.254.97
+    - 54.218.23.75
+
+    As with the previous example, there are no public IP addresses available for the private nodes. 
+
+## Formatting API output using jq
```
Comment (jkoelker) [inline pages/1.13/administering-clusters/locate-public-agent/index.md]: In concert with my above note, perhaps move this earlier in the doc, show both an formatted and formatted response and make the note that from here on out all responses have been formatted for readability.

File: pages/1.13/administering-clusters/locate-public-agent/index.md [hunk 4]
```diff
@@ -6,23 +6,193 @@ menuWeight: 3
 excerpt: Finding a public agent IP address.
 enterprise: false
 ---
+After you have installed DC/OS with a public agent node declared, you can navigate to the public IP address of your public agent node. You can expose the public-facing IP address for an agent as a gateway for access to services that are running inside the DC/OS cluster. For example, if you are configuring load balancing to distribute inbound requests to the services in a cluster, the requests are typically routed through the public IP address frontend to an appropriate service instance backend isolated behind a firewall. 
 
-After you have installed DC/OS with a public agent node declared, you can navigate to the public IP address of your public agent node.
+# Before you begin
+- You must have DC/OS installed with at least one master node and at least one [public agent](/1.13/overview/concepts/#public-agent-node) node.
+- You must have the most recent version of the DC/OS [CLI](/1.13/cli/) installed.
+- You must have the [jq](https://github.com/stedolan/jq/wiki/Installation) language installed if you want to format the output from an API call.
```
Comment (jkoelker) [inline pages/1.13/administering-clusters/locate-public-agent/index.md]: I've read too many RFC and `must` means its mandatory ;).

```suggestion
- You may use [jq](https://github.com/stedolan/jq/wiki/Installation) if you want to format the output from an API call.
```

File: pages/1.13/administering-clusters/locate-public-agent/index.md [hunk 5]
```diff
@@ -6,23 +6,193 @@ menuWeight: 3
...
+
+1. Review the command output.
+
+    This API call returns information similar to the following in an unfiltered format:
+
+    ```bash
+    [{"updated":"2019-01-07T22:22:22.171Z","public_ips":["34.212.37.79"],"private_ip":"10.0.6.210","hostname":"ip-10-0-6-210"},{"updated":"2019-01-07T22:22:22.119Z","public_ips":["52.25.254.97"],"private_ip":"10.0.6.181","hostname":"ip-10-0-6-181"},{"updated":"2019-01-07T22:21:09.585Z","public_ips":["54.218.23.75"],"private_ip":"10.0.6.148","hostname":"ip-10-0-6-148"},{"updated":"2019-01-07T22:22:28.582Z","public_ips":[],"private_ip":"10.0.1.139","hostname":"ip-10-0-1-139"},{"updated":"2019-01-07T22:22:28.649Z","public_ips":[],"private_ip":"10.0.0.138","hostname":"ip-10-0-0-138"}]
+    ```
+
+    In this example, there are three IP addresses for the public agents:
```
Comment (jkoelker) [inline pages/1.13/administering-clusters/locate-public-agent/index.md]: One of these IPs is on the master node (assuming this was from a 1 master and 2 public, 2 private agent setup). Not sure really how to phrase this correctly, but technically not all the ips are on public agents.

File: pages/1.13/administering-clusters/locate-public-agent/index.md (hunk 2 above)
Comment (lsgunnlsgunn) [reply pages/1.13/administering-clusters/locate-public-agent/index.md]: good idea

File: pages/1.13/administering-clusters/locate-public-agent/index.md (hunk 3 above)
Comment (lsgunnlsgunn) [reply pages/1.13/administering-clusters/locate-public-agent/index.md]: great suggestion.

File: pages/1.13/administering-clusters/locate-public-agent/index.md (hunk 4 above)
Comment (lsgunnlsgunn) [reply pages/1.13/administering-clusters/locate-public-agent/index.md]: fixed.

**Pull request merged**